### PR TITLE
Delete derivatives when FileSet is destroyed.

### PR DESCRIPTION
**Pull request by tpendragon**

### CODE REVIEW

File: curation_concerns-models/app/services/curation_concerns/derivative_path.rb [hunk 1]
```diff
@@ -7,8 +7,19 @@ def derivative_path_for_reference(object, destination_name)
...
+      # Returns all derivatives on the filesystem for this object.
+      def derivatives_for_reference(object)
+        Dir.glob(root_path(object).join("*")).select do |path|
+          path.start_with?(derivative_path(object, "", ""))
+        end
+      end
+
       private
 
+        def root_path(object)
```
Comment (jcoyne) [inline curation_concerns-models/app/services/curation_concerns/derivative_path.rb]: Can you add a comment about what this method does?

File: curation_concerns-models/app/services/curation_concerns/derivative_path.rb [hunk 2]
```diff
@@ -7,8 +7,19 @@ def derivative_path_for_reference(object, destination_name)
         derivative_path(object, extension_for(destination_name), destination_name)
       end
 
+      # Returns all derivatives on the filesystem for this object.
```
Comment (jcoyne) [inline curation_concerns-models/app/services/curation_concerns/derivative_path.rb]: The derivatives themselves, or a list of paths?

File: curation_concerns-models/app/services/curation_concerns/derivative_path.rb [hunk 3]
```diff
@@ -7,8 +7,19 @@ def derivative_path_for_reference(object, destination_name)
...
+      def derivatives_for_reference(object)
+        Dir.glob(root_path(object).join("*")).select do |path|
+          path.start_with?(derivative_path(object, "", ""))
```
Comment (jcoyne) [inline curation_concerns-models/app/services/curation_concerns/derivative_path.rb]: can `derivative_path(object, "", "")` be factored out.

File: curation_concerns-models/app/services/curation_concerns/derivative_path.rb [hunk 4]
```diff
@@ -7,8 +7,19 @@ def derivative_path_for_reference(object, destination_name)
...
+        end
+      end
+
       private
 
+        def root_path(object)
+          Pathname.new(derivative_path(object, "", "")).dirname
```
Comment (jcoyne) [inline curation_concerns-models/app/services/curation_concerns/derivative_path.rb]: here's the other place we're using `derivative_path(object, "", "")`

File: curation_concerns-models/app/services/curation_concerns/derivative_path.rb (hunk 3 above)
Comment (grosscol) [inline curation_concerns-models/app/services/curation_concerns/derivative_path.rb]: What do you think about having `""` be the default value for those two args in `def derivative_path` ?

**Comment by tpendragon** [comment]
@jcoyne Take another look?

**Pull request merged**